pkg/forwarder/forwarder.go

```diff
@@ -298,26 +315,26 @@ func (w *Worker) Run(ctx context.Context) {
 func (w *Worker) forward(ctx context.Context) error {
 	w.lock.Lock()
 	defer w.lock.Unlock()
+	fStart := time.Now()
 
-	// Load the match rules each time.
-	from := w.from
-
-	// reset query from last invocation, otherwise match rules will be appended
-	w.from.RawQuery = ""
-	v := from.Query()
-	for _, rule := range w.rules {
-		v.Add("match[]", rule)
+	families, err := w.getFederateMetrics(ctx)
+	if err != nil {
+		statusErr := w.status.UpdateStatus("Degraded", "Degraded", "Failed to retrieve metrics")
+		if statusErr != nil {
+			rlogger.Log(w.logger, rlogger.Warn, "msg", failedStatusReportMsg, "err", err)
+		}
+		return err
 	}
-	from.RawQuery = v.Encode()
 
-	req := &http.Request{Method: "GET", URL: from}
-	families, err := w.fromClient.Retrieve(ctx, req)
+	rfamilies, err := w.getRecordingMetrics(ctx)
 	if err != nil {
-		statusErr := w.status.UpdateStatus("Degraded", "Degraded", "Failed to retrieve metrics")
+		statusErr := w.status.UpdateStatus("Degraded", "Degraded", "Failed to retrieve recording metrics")
```

Comment:
if failed to get result for recording rule, should we return an error, or return the federate result only?
I guess latter?

Reply:
fixed, won't return error when getRecordingMetrics has error.